improver/nbhood/recursive_filter.py

```diff
@@ -366,38 +307,17 @@ def _set_smoothing_coefficients(
                 Cube containing a padded array of smoothing_coefficient values
                 for the specified direction.
         """
+        # The smoothing coefficients cube is expected to be shorter than
+        # the input cube by one cell in one of the x or y dimensions.
```

Comment:
While this is true, I think we should be checking that the coord.points are the mid-points of the equivalent cube.coord too.

Reply:
I've added more checking here and altered the unit test inputs, so that they're more representative of the expected smoothing coefficient inputs i.e. matching the outputs from the `generate_orographic_smoothing_coefficients` CLI.